internal/transport/http2_server.go

```diff
@@ -402,6 +403,20 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 		return true
 	}
 	t.maxStreamID = streamID
+	if state.data.httpMethod != http.MethodPost {
```

Comment:
Not sure why the comments on the previous PR are missing. But didn't @dfawley say something to the tune of moving this up before the stream ID odd number check?

Reply:
I had to open a new branch because the old PR was a branch on this repository.

Reply:
No, the request was to move it after this line so the max stream ID persists in the http2_server.